src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php

```diff
@@ -162,7 +162,15 @@ private static function addChoiceView($choice, string $value, $label, array $key
         // $value may be an integer or a string, since it's stored in the array
         // keys. We want to guarantee it's a string though.
         $key = $keys[$value];
-        $nextIndex = \is_int($index) ? $index++ : $index($choice, $key, $value);
+        if (\is_int($index)) {
+            $nextIndex = (string) $index++;
+        } else {
+            $nextIndex = $index($choice, $key, $value);
+
+            if (!\is_string($nextIndex)) {
+                trigger_deprecation('symfony/form', '6.0.0', 'The "choice_name" callback returns "%s", return "string" instead.', \gettype($nextIndex));
```

Comment:
Since you're targeting the 6.0 branch: We don't add deprecations in a bugfix release.